components/storage_types/src/region_info.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use kvproto::metapb::Region;
+use raft::StateRole;
+use std::error;
+
+#[derive(Clone, Debug)]
+pub struct RegionInfo {
+    pub region: Region,
+    pub role: StateRole,
+}
+
+impl RegionInfo {
+    pub fn new(region: Region, role: StateRole) -> Self {
+        Self { region, role }
+    }
+}
+
+pub type SeekRegionCallback = Box<dyn Fn(&mut dyn Iterator<Item = &RegionInfo>) + Send>;
+
+pub trait RegionInfoProvider: Send + Clone + 'static {
```

Comment:
@MyonKeminta do you think it is fine to let the storage contain Region Info? or do we need to move this to another place?

Reply:
I believe we should find a better place for it, since it knows Regions, which should be in storage. It was my fault to put it in the storage module.

Reply:
Maybe in raftstore, I think

Reply:
Would you like this to happen in this PR or a later PR?

Reply:
As currently factored, the storage gc depends on this trait, so it can't live in raftstore while also making storage not depend on raftstore (which is the goal of this crate of mutual dependencies). I have not looked the storage gc yet, and would rather not try to refactor that now.

Reply:
I pushed a commit that adds a fixme comment to this file.

Reply:
Ok. It seems this should be done after extracting GCWorker.